src/KafkaFlow.Retry/Durable/Definitions/Builders/Polling/PollingDefinitionsAggregatorBuilder.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace KafkaFlow.Retry
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using KafkaFlow.Retry.Durable.Definitions.Polling;
+
+    public class PollingDefinitionsAggregatorBuilder
+    {
+        private readonly List<PollingDefinition> pollingDefinitions;
+        private string schedulerId;
+
+        public PollingDefinitionsAggregatorBuilder()
+        {
+            this.pollingDefinitions = new List<PollingDefinition>();
+        }
+
+        public PollingDefinitionsAggregatorBuilder WithCleanupPollingConfiguration(Action<CleanupPollingDefinitionBuilder> configure)
+        {
+            var cleanupPollingDefinitionBuilder = new CleanupPollingDefinitionBuilder();
+            configure(cleanupPollingDefinitionBuilder);
+            var cleanupPollingDefinition = cleanupPollingDefinitionBuilder.Build();
+
+            this.pollingDefinitions.Add(cleanupPollingDefinition);
+
+            return this;
+        }
+
+        public PollingDefinitionsAggregatorBuilder WithRetryDurablePollingConfiguration(Action<RetryDurablePollingDefinitionBuilder> configure)
+        {
+            var retryDurablePollingDefinitionBuilder = new RetryDurablePollingDefinitionBuilder();
+            configure(retryDurablePollingDefinitionBuilder);
+            var retryDurablepollingDefinition = retryDurablePollingDefinitionBuilder.Build();
+
+            this.pollingDefinitions.Add(retryDurablepollingDefinition);
+
+            return this;
+        }
+
+        public PollingDefinitionsAggregatorBuilder WithSchedulerId(string schedulerId)
+        {
+            this.schedulerId = schedulerId;
+            return this;
+        }
+
+        internal PollingDefinitionsAggregator Build()
+        {
+            this.FillMissingOptionalPollingDefinitions();
+
+            return new PollingDefinitionsAggregator(this.schedulerId, this.pollingDefinitions);
```

Comment:
Just to think: is there any negative consequence of calling `Build()` without any polling definition configured?

If I call:

```csharp
b.WithSchedulerId("test")
    .Build();
```

A cleanup polling policy is added, but there is no retry durable polling policy. Should we allow this?

Reply:
Should we also allow a null or empty `schedulerId`? Also related to comment on line 40.

Reply:
We are validating the `schedulerId` in the `PollingDefinitionsAggregator` constructor, which is called in the `Build` method.

If we call:
```
b.WithSchedulerId("test")
    .Build();
```

An exception will be thrown because the retry durable polling job is required. This validation is called in the `Build` method as well.